### ISSUE TITLE
Template Library

### ISSUE THREAD

**Issue by codewithkyle** [codewithkyle/craftcms-boilerplate#24]
### What Is It?
Like the block library the template library should be demo pages built using the page build matrix field. Content editors can browse and select a template as a starting point. Initially only developers should be able to create templates with the possibility of expanding the functionality into a full template builder.

### Why We Need It?
Sometimes client's block libraries get extremely large and content editors don't look at every block before building a page. This can lead to development time being spent on a block no one knows about or content editors requesting blocks that already exists. Also some clients want a one-click landing page generator where all they have to do is fill in content.

Dependent on #22 and #23

**Comment by codewithkyle** [comment]
No longer needed. See #29
